### ISSUE TITLE
Add tags to spawned objects by default

### ISSUE THREAD

**Issue by kulzae** [Neos-Metaverse/NeosPublic#3773]
### Is your tweak request related to a problem? Please describe.

User spawned videos and images commonly litter worlds and currently are named whatever the file or inputting URL is.

### Describe what would you like tweaked

Id like these to be tagged with a tag to help identify them or separate them from the rest of the world so they can be cleaned up or organized automatically. Images could be tagged as "uploaded image" or something similar while videos could be "video". These tags could be removed upon saving so they are only "temporary"

### Describe alternatives you've considered

searching the hierarchy of the user slot parent for anything that wasn't there before but this does not work for things that end up in root or somewhere else. it can also end up grabbing spawned items from inventory

**Comment by BlueCyro** [comment]
I have to agree that this'd be really useful. People are supposed to use tags anyways to make custom identifiers for objects they make, it only makes sense to have the same for uniquitous objects like video players and screenshots, or even audio streams.

**Comment by epicEaston197** [comment]
in order for this to be remotely viable the tags would have to be added to already existing objects so if you have a item that did not have a tag before and you spawned it out it would add the tag afterwards

**Comment by BlueCyro** [comment]
> in order for this to be remotely viable the tags would have to be added to already existing objects so if you have a item that did not have a tag before and you spawned it out it would add the tag afterwards

A lot of these items are created and re-created on a sessionly basis. I think it's a little dramatic to say it wouldn't be 'remotely viable' since 90% of the aforementioned items (pictures, video players, audio streams, etc.) are all very actively created. Doesn't seem infeasible with some upgrade code.

**Comment by JackTheFoxOtter** [comment]
I suggested #3627 a while ago which might be a potential partial solution for this issue, as it would introduce a node that gives you access to an event that fires when an upload has been completed. It would also fire after any processing so you don't have to worry about it not having finished the upload process before doing any changes to it. You'd also know which user uploaded it, allowing you to easily move it into an organization slot should you want to.

The problem with adding tags automatically is that many existing LogiX systems rely on certain tags to be present. While this might work well for video players or image previews, it could cause issues for other items. I'm also not sure if this could affect cloud spawning, even if not actively supported at the current date it should probably be considered.

That said tho, I would LOVE a better way of detecting what kind of object an item in the world is. In many cases currently you need to do a bunch of "guessing" as without component level access you can't determine what type of asset (picture, UI, mesh, video player, gizmo etc.) an object is without relying on heuristics based on attributes you have access to like names or dimensions / slot structure, which I believe shouldn't be relied upon as this is information that may change in a future update. I'm not sure if the current tag system is a suitable solution for this issue tho. Perhaps a separate "asset type" information could be introduced that has a clear list of possible values to circumvent ambiguity issues?
